tasks/gen/action_generator.cr

```diff
@@ -1,24 +1,26 @@
-require "lucky_cli"
-require "teeplate"
 require "colorize"
 require "file_utils"
 
 class Lucky::ActionTemplate < Teeplate::FileTree
   @name : String
   @action : String
+  @inherit_from : String
 
   directory "#{__DIR__}/templates/action"
 
-  def initialize(@name, @action)
+  def initialize(@name, @action, @inherit_from)
   end
 end
 
-class Gen::Action < LuckyCli::Task
+class Gen::ActionGenerator < LuckyCli::Task
   banner "Generate a new action"
 
-  def call(io : IO = STDOUT)
+  def call
+  end
```

Comment:
To get around this I believe you can make this class `abstract` That means that it can't be instantiated by itself, and that it doesn't need to implement the methods yet

`abstract class Gen::...`

Reply:
I think that because it inherits from `LuckyCli::Task` it gets instantiated automatically when `tasks.cr` requires the tasks. So when I make it an abstract class it spits this error out:

```
Error in line 1: while requiring "./spec/http_method_override_handler_spec.cr"

in spec/http_method_override_handler_spec.cr:1: while requiring "./spec_helper"

require "./spec_helper"
^

in spec/spec_helper.cr:3: while requiring "../tasks/**"

require "../tasks/**"
^

in tasks/gen/action_generator.cr:15: expanding macro

abstract class Gen::ActionGenerator < LuckyCli::Task
^

in macro 'inherited' /home/mikmak/crystal/lib/lucky/lib/lucky_cli/src/lucky_cli/task.cr:2, line 1:

>  1.     LuckyCli::Runner.tasks << self.new
   2. 
   3.     def name
   4.       "gen.action_generator"
   5.     end
   6.   

instantiating 'Gen::ActionGenerator:Class#new()'
in /usr/share/crystal/src/primitives.cr:36: can't instantiate abstract class Gen::ActionGenerator

class Reference
^
```

Reply:
Ah I see. That is something that should change in LuckyCLI.

This line:
https://github.com/luckyframework/lucky_cli/blob/master/src/lucky_cli/task.cr#L3

Should only be run if the type is *not* abstract, since abstract classes can't be instantiated. I'll add an issue.

I think for now to work around this you could instead make that class not inherit from `LuckyCLI::Task` and instead call it from within the generators like this: `Lucky::ActionGenerator.new(action_name : String, inherit_from : String).generate`

Does that make sense?

Reply:
@mikeeus I think this is the only part that should change. Other than that, this looks great! The reason it shouldn't inherit from LuckyCLI is that it will be listed as a task when running `lucky --help`, even though you can't run it. So extracting it to it's own class would fix that